### PR TITLE
Tests for JerryProtocolHandler

### PR DESCRIPTION
**Pull request by Yuyupo**
Add tests for exiting/new functions

IoT.js-VSCode-DCO-1.0-Signed-off-by: Daniella Barsony bella@inf.u-szeged.hu

### CODE REVIEW

File: src/test/JerryProtocolHandler.test.ts [hunk 1]
```diff
@@ -590,5 +603,106 @@ suite('JerryProtocolHandler', () => {
...
+          assert(debugClient.send.withArgs(Uint8Array.from([SP.CLIENT.JERRY_DEBUGGER_CONTINUE])));
+        });
+
+        test('pause() throwing error when pausing at breakpoint', async () => {
+          const { handler } = setupHaltedProtocolHandler(true);
+          await handler.pause()
+                .catch(error => {
+                    assert.strictEqual((<Error>error).message, 'attempted pause while at breakpoint');
+                  });
+        });
```
Comment (knightburton) [inline src/test/JerryProtocolHandler.test.ts]: The indentation is a bit large. Put the `.catch` 2 space away from the await.

File: src/test/JerryProtocolHandler.test.ts [hunk 2]
```diff
@@ -590,5 +603,106 @@ suite('JerryProtocolHandler', () => {
...
+      });
+    });
+
+    suite('sendClientSourceControl()', () => {
+      test('throws if index is -1', async () => {
+        const handler = new JerryDebugProtocolHandler({});
+        await handler.sendClientSourceControl(-1)
+                .catch(error => {
+                    assert.strictEqual((<Error>error).message, 'Invalid source sending control code.');
+                  });
```
Comment (knightburton) [inline src/test/JerryProtocolHandler.test.ts]: Same as above.

File: src/test/JerryProtocolHandler.test.ts [hunk 3]
```diff
@@ -590,5 +603,106 @@ suite('JerryProtocolHandler', () => {
...
+      });
+    });
+
+    suite('sendClientSource()', () => {
+      test('throws if waitForSource is not enabled', async () => {
+        const handler = new JerryDebugProtocolHandler({});
+        await handler.sendClientSource('onelittlekitten', 'and some more')
+                .catch(error => {
+                    assert.strictEqual((<Error>error).message, 'wait-for-source not enabled');
+                  });
```
Comment (knightburton) [inline src/test/JerryProtocolHandler.test.ts]: Same

**Review by knightburton — CHANGES_REQUESTED** [review]

**Review by knightburton — APPROVED** [review]
LGTM

**Review by yichoi — APPROVED** [review]
LGTM

**Pull request merged**